tabular/src/autogluon/tabular/models/catboost/catboost_model.py

```diff
@@ -178,88 +224,96 @@ def _fit(self, X_train, y_train, X_val=None, y_val=None, time_limit=None, num_gp
-            available_mem = psutil.virtual_memory().available
-            if self.problem_type == SOFTCLASS:  # TODO: remove this once catboost-dev is no longer necessary and SOFTCLASS objectives can be pickled.
-                model_size_bytes = 1  # skip memory check
+                params_final['iterations'] = min(params['iterations'] - num_sample_iter, estimated_iters_in_time)
+                if params_final['iterations'] > max_memory_iters - num_sample_iter:
+                    if max_memory_iters - num_sample_iter <= 500:
+                        logger.warning('\tWarning: CatBoost will be early stopped due to lack of memory, increase memory to enable full quality models, max training iterations changed to %s from %s' % (max_memory_iters, params_final['iterations'] + num_sample_iter))
+                    params_final['iterations'] = max_memory_iters - num_sample_iter
             else:
-                model_size_bytes = sys.getsizeof(pickle.dumps(self.model))
+                params_final = params.copy()
 
-            max_memory_proportion = 0.3 * max_memory_usage_ratio
-            mem_usage_per_iter = model_size_bytes / num_sample_iter
-            max_memory_iters = math.floor(available_mem * max_memory_proportion / mem_usage_per_iter)
-
-            params_final['iterations'] = min(params['iterations'] - num_sample_iter, estimated_iters_in_time)
-            if params_final['iterations'] > max_memory_iters - num_sample_iter:
-                if max_memory_iters - num_sample_iter <= 500:
-                    logger.warning('\tWarning: CatBoost will be early stopped due to lack of memory, increase memory to enable full quality models, max training iterations changed to %s from %s' % (max_memory_iters, params_final['iterations'] + num_sample_iter))
-                params_final['iterations'] = max_memory_iters - num_sample_iter
+            if train_dir is not None:
+                params_final['train_dir'] = train_dir
+            if params_final['iterations'] > 0:
+                init_model = self.model
+
+                self.model = model_type(
+                    **params_final,
+                )
+
+                # TODO: Strangely, this performs different if clone init_model is sent in than if trained for same total number of iterations. May be able to optimize catboost models further with this
+                self.model.fit(
+                    X_train,
+                    eval_set=eval_set,
+                    verbose=verbose,
+                    early_stopping_rounds=early_stopping_rounds,
+                    # use_best_model=True,
+                    init_model=init_model,
+                )
+
+                self._merge_prev_model(init_model, metric_name)
         else:
+            #TODO: Time limit constraint is not considered in warm starting scenairo. May need to add it
+            self._prev_model = self.model
```

Comment:
why are we storing the previous model? What is the need to have 2 models stored in this object at the same time?

Reply:
For incremental training with multi-fidelity training. So that additional trees could be easily added to the current model

Reply:
I don't believe this is necessary, please refer to other comment.